ts/main.ts

```diff
@@ -0,0 +1,117 @@
+) as HTMLParagraphElement;
+const marketCapLabel = document.querySelector(
+  '.market-cap',
+) as HTMLHeadingElement;
+const dataMarketCapUsd = document.getElementById(
+  'dataMarketCapUsd',
+) as HTMLHeadingElement;
+
+// Function to fetch crypto data by name and set it to lower case for API requirements
+const apiUrl = 'https://api.coincap.io/v2/assets';
+
+async function fetchCryptoData(name: string): Promise<Asset | null> {
+  try {
+    const response = await fetch(`${apiUrl}/${name.toLowerCase()}`);
+    const data = await response.json();
+    if (data && data.data) {
+      return {
+        name: data.data.name,
+        symbol: data.data.symbol,
+        priceUsd: parseFloat(data.data.priceUsd),
+        changePercent24Hr: parseFloat(data.data.changePercent24Hr),
+        marketCapUsd: parseFloat(data.data.marketCapUsd),
+      };
+    }
+  } catch (error) {
+    console.error('Error fetching crypto data:', error);
+  }
+  return null;
+}
+
+// Event listener for the search button click
+searchButton.addEventListener('click', async () => {
+  const name = searchInput.value.trim();
+  if (name) {
+    const asset = await fetchCryptoData(name);
+    if (asset) {
+      // Display the fetched asset data
+      dataName.textContent = asset.name;
+      dataSymbol.textContent = asset.symbol;
+      dataPriceUsd.textContent = `$${asset.priceUsd.toFixed(2)}`;
+      dataChangePercent24Hr.textContent = `${asset.changePercent24Hr.toFixed(2)}%`;
+
+      // Set color classes
+      dataChangePercent24Hr.classList.remove('positive', 'negative');
+
+      // Apply correct class based on value
+      if (asset.changePercent24Hr > 0) {
+        dataChangePercent24Hr.textContent = `+${asset.changePercent24Hr.toFixed(2)}% (24Hr)`;
+        dataChangePercent24Hr.classList.add('positive');
+      } else {
+        dataChangePercent24Hr.textContent = `${asset.changePercent24Hr.toFixed(2)}% (24Hr)`;
+        dataChangePercent24Hr.classList.add('negative');
+      }
+
+      // Convert and display the market cap in billions
+      marketCapLabel.textContent = 'Market Cap';
+      dataMarketCapUsd.textContent = `$${(asset.marketCapUsd / 1_000_000_000).toFixed(2)} B`;
+
+      // Display the data section
+      cryptoDataDiv.style.display = 'block';
+
+      // Create the Add to Watchlist button dynamically
+      const addToWatchlistBtn = document.createElement('button');
+      addToWatchlistBtn.textContent = 'Add to Watchlist';
+      addToWatchlistBtn.classList.add('add-to-watchlist');
+
+      // Remove any existing "Add to Watchlist" button before appending a new one
+      const existingBtn = cryptoDataDiv.querySelector('.add-to-watchlist');
+      if (existingBtn) {
+        existingBtn.remove();
+      }
+      addToWatchlistBtn.addEventListener('click', () => {
+        console.log('Add to watchlist clicked!');
+
+        // // Save the asset to the watchlist using the `addAsset` function from `data.ts`
+        // addAsset(asset);
```

Comment:
Remove all commented out code. I know you may use this in a future feature, but if it's not being used now, it shouldn't be in your code